msgpack/fallback.py

```diff
@@ -715,6 +715,20 @@ def _pack(self, obj, nest_limit=DEFAULT_RECURSE_LIMIT,
                 else:
                     raise PackValueError("String is too large")
                 return self._buffer.write(obj)
+            if check(obj, memoryview):
+                n = len(obj) * obj.itemsize
+                if self._use_bin_type:
+                    if n <= 0xff:
+                        self._buffer.write(struct.pack('>BB', 0xc4, n))
+                    elif n <= 0xffff:
+                        self._buffer.write(struct.pack(">BH", 0xc5, n))
+                    elif n <= 0xffffffff:
+                        self._buffer.write(struct.pack(">BI", 0xc6, n))
+                    else:
+                        raise ValueError("memoryview is too large")
+                    return self._buffer.write(obj)
+                else:
+                    raise ValueError("need use_bin_type=True to pack memoryview")
```

Comment:
Still not getting it.

If you have bytes object and the content happens to be validly encoded bytes, it will happily decode them. If not, then not.

If you have a memoryview of such a bytes object and the contents happens to be validly encoded bytes, it will also happily decode them. If not, then not.

So, what's the fundamentaly difference?